### ISSUE TITLE
Constrain which templates/zones a block can go into

### ISSUE THREAD

**Issue by chiel** [chiel/omni-old#14]
Blocks might have to be restructured a bit to get rid of having to repeat metadata. A block could perhaps just be a function with some attributes, which calls the actual block constructor. For example:

``` js
'use strict';

var meta = {
    name: 'Gallery',
    description: 'Header gallery to showcase content',
    template: /.*/,
    zones: /^header$/
};

var Gallery = function(){};

var block = function(){
    return new Gallery();
};

block.meta = meta;
module.exports = block;
```

The above example would create a block that can be dropped on any template, but only in a zone that is called `header`. It is probably most convenient if `template` and `zones` are regular expressions so you can do partial matches, or match multiple specific templates.

Or, there could be a `constrain` property which takes the form of `<template>.<zone>` or so. Should experiment to see what is more convenient.

**Comment by chiel** [comment]
Part of this feature should already be in v1, namely the restructuring of the blocks (so that it's a function that creates the actual block) otherwise backwards compatibility will already be instantly broken.

**Comment by chiel** [comment]
Step 1 of this has been implemented for v1 in 89c44cbfbaf72d84b9f95440f91492c9c60ad674.
